File: src/main/java/net/hectus/neobb/turn/default_game/block/TGreenCarpet.java
package net.hectus.neobb.turn.default_game.block;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.default_game.CounterFilter;
import net.hectus.neobb.turn.default_game.attributes.clazz.NeutralClazz;
import net.hectus.neobb.turn.default_game.attributes.function.CounterattackFunction;
import org.bukkit.block.Block;

import java.util.List;

public class TGreenCarpet extends BlockTurn implements CounterattackFunction, NeutralClazz {
    public TGreenCarpet(NeoPlayer player) { super(player); }
    public TGreenCarpet(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public int cost() {
        return 3;
    }

    @Override
    public List<CounterFilter> counters() {
        return List.of(CounterFilter.of(turn -> turn.getClass().getSimpleName().endsWith("Wool"), "wool"));
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/default_game/block/TGreenCarpet.kt
package net.hectus.neobb.modes.turn.default_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.modes.turn.default_game.CounterFilter
import net.hectus.neobb.modes.turn.default_game.attribute.clazz.NeutralClazz
import net.hectus.neobb.modes.turn.default_game.attribute.function.CounterattackFunction
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block

class TGreenCarpet(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), CounterattackFunction, NeutralClazz {
    override val cost: Int = 3

    override fun counters(): List<CounterFilter> {
        return listOf(CounterFilter.of("wool") { it::class.simpleName!!.endsWith("Wool") })
    }
}
